### ISSUE TITLE
Dependency conflict (bleak) with rivian-python-client on HA 2024.6.0b0

### ISSUE THREAD

**Issue by krazos** [bretterer/home-assistant-rivian#161]
HA 2024.6.0b0 requires `bleak==0.22.1` which results in a dependency conflict with rivian-python-client, for which bleak is supposedly optional:

```
[tool.poetry.dependencies]
python = ">=3.9, <3.13"
aiohttp = ">=3.0.0"
cryptography = ">=41.0.1, <43.0"
backports-strenum = { version = "^1.2.4", python = "<3.11" }
bleak = { version = "^0.21", optional = true }
dbus-fast = {version = "^2.11.0", optional = true, platform = "linux"}
```

However, rivian-python-client is uninstalled as part of the update to core-2024.6.0b0. I had to manually install rivian-python-client in the HA Docker container, which downgraded bleak to 0.21.1. I then manually upgraded bleak back to 0.22.1 and, after patching the TypeVar issue for which I created a separate PR, everything is back up and running smoothly.

**Comment by natekspencer** [comment]
Thanks for raising this. While bleak is optional for the base rivian-python-client, we do require it in the Home Assistant integration to allow for vehicle pairing to enable control. I unfortunately don't know a way to dynamically require it based on the configuration.

That being said, we do need to update the bleak dependency constraints in that project so I'll get that done!

**Comment by natekspencer** [comment]
This should be resolved with [release 1.2.1](https://github.com/bretterer/home-assistant-rivian/releases/tag/1.2.1)
